sdk/tables/data-tables/src/TableBatch.ts

```diff
@@ -0,0 +1,294 @@
+    this.pendingOperations.push(this.interceptClient.updateEntity(entity, mode, options));
+  }
+
+  /**
+   * Submits the operations in the batch
+   */
+  public async submitBatch(): Promise<any> {
+    await Promise.all(this.pendingOperations);
+    const body = this.batchRequest.getHttpRequestBody();
+    const client = new ServiceClient();
+    const headers: RawHttpHeaders = {
+      accept: "application/json",
+      "x-ms-version": "2019-02-02",
+      "Accept-Charset": "UTF-8",
+      DataServiceVersion: "3.0;",
+      MaxDataServiceVersion: "3.0;NetFx",
+      "Content-Type": `multipart/mixed; boundary=batch_${this.batchGuid}`,
+      Connection: "Keep-Alive"
+    };
+
+    const { span, updatedOptions } = createSpan("TableBatch-submitBatch", {});
+    const request = new WebResource(this.url, "POST", body, undefined, new HttpHeaders(headers));
+
+    request.spanOptions = updatedOptions;
+
+    if (this.credential) {
+      const authHeader = getAuthorizationHeader(request, this.credential);
+      request.headers.set("Authorization", authHeader);
+    }
+
+    const requestOptions: RequestPrepareOptions = {
+      method: "POST",
+      url: this.url,
+      headers: request.headers.rawHeaders(),
+      body,
+      disableJsonStringifyOnBody: true
+    };
+    try {
+      const rawBatchResponse = await client.sendRequest(requestOptions);
+      return parseBatchResponse(rawBatchResponse);
+    } catch (error) {
+      span.setStatus({
+        code: CanonicalCode.UNKNOWN,
+        message: error.message
+      });
+      throw error;
+    } finally {
+      span.end();
+    }
+  }
+
+  private checkPartitionKey(partitionKey: string): void {
+    if (this.partitionKey !== partitionKey) {
+      throw new Error("All operations in a batch must target the same partitionKey");
+    }
+  }
+}
+
+function parseBatchResponse(response: HttpOperationResponse): TableBatchResponse {
+  const subResponsePrefix = `--changesetresponse_`;
+  const status = response.status;
+  const rawBody = response.bodyAsText || "";
+  const splitBody = rawBody.split(subResponsePrefix); // string after ending is useless
+  const subResponses = splitBody.slice(1, splitBody.length - 1); // string before first response boundary is useless
+
+  const responses: TableBatchEntityResponse[] = subResponses.map((response) => {
+    const statusMatch = response.match(/HTTP\/1.1 ([0-9]*)/);
+    const etagMatch = response.match(/ETag: (.*)/);
+    const rowKeyMatch = response.match(/RowKey='(.*)'/);
+    const bodyMatch = response.match(/\{(.*)\}/);
+    const body = bodyMatch?.length === 2 ? JSON.parse(`${bodyMatch[0]}`) : undefined;
+
+    if (statusMatch?.length !== 2) {
+      throw new Error(`Couldn't extract status from sub-response:\n ${response}`);
+    }
+
+    const status = Number.parseInt(statusMatch[1]);
+    if (!Number.isInteger(status)) {
+      throw new Error(`Expected sub-response status to be an integer ${status}`);
+    }
```

Comment:
should we do these checks on status immediately after matching it so we don't waste effort matching the other headers or calling JSON.parse on the body?